### PR TITLE
refactor: chain detailed

### PR DESCRIPTION
**Pull request by guanbinrui**
closes #

### CODE REVIEW

File: packages/maskbook/src/plugins/RedPacket/SNSAdaptor/hooks/useAvailabilityComputed.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ export function useAvailabilityComputed(account: string, payload: RedPacketJSONP
     const isClaimed = availability.claimed_amount ? availability.claimed_amount !== '0' : availability.ifclaimed
     const isRefunded = isEmpty && Number.parseInt(availability.claimed, 10) < Number.parseInt(availability.total, 10)
     const isCreator = isSameAddress(payload?.sender.address ?? '', account)
-    const parsedChainId = payload.network ? getChainIdFromName(payload.network) : ChainId.Mainnet
+    const parsedChainId = getChainIdFromName(payload.network ?? '') ?? ChainId.Mainnet
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/RedPacket/SNSAdaptor/hooks/useAvailabilityComputed.ts:33]: cc @zhouhanseng

File: packages/maskbook/src/plugins/Wallet/services/assets.ts [hunk 2]
```diff
@@ -101,32 +101,28 @@ export async function getAssetsList(
...
-            value: {
-                [CurrencyType.USD]: new BigNumber(x.price)
-                    .multipliedBy(new BigNumber(x.balance).dividedBy(pow10(x.decimals)))
-                    .toFixed(),
-            },
-            logoURL: x.logo_url,
-        }
-    })
+    return data
+        .filter((x) => getChainIdFromName(x.id))
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Wallet/services/assets.ts:105]: Filter not supported tokens out.

**Review by zhouhanseng — APPROVED** [review]

File: packages/maskbook/src/plugins/ITO/SNSAdaptor/PostInspector.tsx [hunk 3]
```diff
@@ -51,7 +51,7 @@ export function PostInspector(props: PostInspectorProps) {
                 (t) =>
                     ({
                         address: t.address,
-                        type: isSameAddress(t.address, NATIVE_TOKEN_ADDRESS)
+                        type: isSameAddress(t.address, NATIVE_TOKEN_ADDRESS ?? '')
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/ITO/SNSAdaptor/PostInspector.tsx:54]: Why will `NATIVE_TOKEN_ADDRESS` be yield to `undefined`?

Comment (septs) [reply packages/maskbook/src/plugins/ITO/SNSAdaptor/PostInspector.tsx:54]: why not use `isNative`

https://github.com/DimensionDev/Maskbook/blob/41767116e15729dd826d042c6afe4805618e58fc/packages/web3-shared/src/utils/address.ts#L22

**Pull request merged**